TestContainerStartOnDaemonRestart fails when dockerd becomes subreaper

Buildkit sets subreaper to 1 and https://github.com/moby/moby/pull/37620 vendors that code, effectively setting dockerd as a subreaper, which causes problems. This issue is to track it.

Here's how to reproduce:

```
$ git fetch git@github.com:tiborvass/docker debug-subreaper3
$ git checkout FETCH_HEAD
$ make shell
root# SUBREAPER=1 TEST_INTEGRATION_DIR=./integration/container TESTFLAGS='-test.run TestContainerStartOnDaemonRestart -test.v' ./hack/make.sh dynbinary test-integration
...
INFO: Testing against a local daemon
=== RUN   TestContainerStartOnDaemonRestart
=== PAUSE TestContainerStartOnDaemonRestart
=== CONT  TestContainerStartOnDaemonRestart
--- FAIL: TestContainerStartOnDaemonRestart (1.83s)
	daemon.go:289: [d3212059849e4] waiting for daemon to start
	daemon.go:321: [d3212059849e4] daemon started
	daemon.go:289: [d3212059849e4] waiting for daemon to start
	daemon.go:321: [d3212059849e4] daemon started
	daemon_linux_test.go:65: assertion failed: error is not nil: Error response from daemon: mkdir /var/run/docker/containerd/daemon/io.containerd.runtime.v1.linux/moby/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2: file exists: already exists: failed to start test container
	daemon.go:279: [d3212059849e4] exiting daemon
FAIL
...
```

```
root# cat ./bundles/test-integration/d3212059849e4/docker.log
time="2018-08-20T18:16:25.718888702Z" level=warning msg="Error while setting daemon root propagation, this is not generally critical but may cause some functionality to not work or fallback to less desirable behavior" dir=/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root error="error writing file to signal mount cleanup on shutdown: open /tmp/docker-execroot/d3212059849e4/unmount-on-shutdown: no such file or directory"
time="2018-08-20T18:16:25.719226115Z" level=debug msg="Listener created for HTTP on unix (/tmp/docker-integration/d3212059849e4.sock)"
time="2018-08-20T18:16:25.720763152Z" level=debug msg="Golang's threads limit set to 115380"
time="2018-08-20T18:16:25.721626604Z" level=info msg="parsed scheme: \"unix\"" module=grpc
time="2018-08-20T18:16:25.721654792Z" level=info msg="scheme \"unix\" not registered, fallback to default scheme" module=grpc
time="2018-08-20T18:16:25.721771385Z" level=info msg="ccResolverWrapper: sending new addresses to cc: [{unix:///var/run/docker/containerd/docker-containerd.sock 0  <nil>}]" module=grpc
time="2018-08-20T18:16:25.721802980Z" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
time="2018-08-20T18:16:25.721863058Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc42026b800, CONNECTING" module=grpc
time="2018-08-20T18:16:25.721861937Z" level=info msg="parsed scheme: \"unix\"" module=grpc
time="2018-08-20T18:16:25.721923019Z" level=info msg="scheme \"unix\" not registered, fallback to default scheme" module=grpc
time="2018-08-20T18:16:25.723685974Z" level=info msg="ccResolverWrapper: sending new addresses to cc: [{unix:///var/run/docker/containerd/docker-containerd.sock 0  <nil>}]" module=grpc
time="2018-08-20T18:16:25.723747200Z" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
time="2018-08-20T18:16:25.724074163Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc4201b8ec0, CONNECTING" module=grpc
time="2018-08-20T18:16:25.724185149Z" level=debug msg="Using default logging driver json-file"
time="2018-08-20T18:16:25.724230092Z" level=debug msg="[graphdriver] trying provided driver: aufs"
time="2018-08-20T18:16:25.725066441Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc42026b800, READY" module=grpc
time="2018-08-20T18:16:25.725618098Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc4201b8ec0, READY" module=grpc
time="2018-08-20T18:16:25.726012691Z" level=debug msg="processing event stream" module=libcontainerd namespace=plugins.moby
time="2018-08-20T18:16:25.726097678Z" level=debug msg="Initialized graph driver aufs"
time="2018-08-20T18:16:25.732966945Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
time="2018-08-20T18:16:25.733390176Z" level=warning msg="Your kernel does not support swap memory limit"
time="2018-08-20T18:16:25.733561538Z" level=warning msg="Your kernel does not support cgroup rt period"
time="2018-08-20T18:16:25.733595360Z" level=warning msg="Your kernel does not support cgroup rt runtime"
time="2018-08-20T18:16:25.734247993Z" level=debug msg="Max Concurrent Downloads: 3"
time="2018-08-20T18:16:25.734425195Z" level=debug msg="Max Concurrent Uploads: 5"
time="2018-08-20T18:16:25.734457805Z" level=info msg="Loading containers: start."
time="2018-08-20T18:16:25.734553154Z" level=debug msg="Option Experimental: false"
time="2018-08-20T18:16:25.734568813Z" level=debug msg="Option DefaultDriver: bridge"
time="2018-08-20T18:16:25.734578314Z" level=debug msg="Option DefaultNetwork: bridge"
time="2018-08-20T18:16:25.734588244Z" level=debug msg="Network Control Plane MTU: 1500"
time="2018-08-20T18:16:25.734604874Z" level=debug msg="processing event stream" module=libcontainerd namespace=moby
time="2018-08-20T18:16:25.762461870Z" level=warning msg="Running modprobe nf_nat failed with message: ``, error: exec: \"modprobe\": executable file not found in $PATH"
time="2018-08-20T18:16:25.762763102Z" level=warning msg="Running modprobe xt_conntrack failed with message: ``, error: exec: \"modprobe\": executable file not found in $PATH"
time="2018-08-20T18:16:25.762912603Z" level=debug msg="Fail to initialize firewalld: Failed to connect to D-Bus system bus: dial unix /var/run/dbus/system_bus_socket: connect: no such file or directory, using raw iptables instead"
time="2018-08-20T18:16:25.768205357Z" level=debug msg="/sbin/iptables, [--wait -t filter -n -L DOCKER-USER]"
time="2018-08-20T18:16:25.771717136Z" level=debug msg="/sbin/iptables, [--wait -t filter -C DOCKER-USER -j RETURN]"
time="2018-08-20T18:16:25.775161169Z" level=debug msg="/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.777587608Z" level=debug msg="/sbin/iptables, [--wait -D FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.780034401Z" level=debug msg="/sbin/iptables, [--wait -I FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.788217792Z" level=debug msg="/sbin/iptables, [--wait -t filter -n -L DOCKER-USER]"
time="2018-08-20T18:16:25.790356984Z" level=debug msg="/sbin/iptables, [--wait -t filter -C DOCKER-USER -j RETURN]"
time="2018-08-20T18:16:25.791884215Z" level=debug msg="/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.794134660Z" level=debug msg="/sbin/iptables, [--wait -D FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.796827029Z" level=debug msg="/sbin/iptables, [--wait -I FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.800075589Z" level=info msg="Default bridge (docker0) is assigned with an IP address 172.18.0.0/16. Daemon option --bip can be used to set a preferred IP address"
time="2018-08-20T18:16:25.800165306Z" level=debug msg="Allocating IPv4 pools for network bridge (2836ffbe26dad9e6561c82801dce7c4d42067335063075ff77bbf7e3a6db7d25)"
time="2018-08-20T18:16:25.800196407Z" level=debug msg="RequestPool(LocalDefault, 172.18.0.0/16, , map[], false)"
time="2018-08-20T18:16:25.800284797Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, 172.18.0.1, map[RequestAddressType:com.docker.network.gateway])"
time="2018-08-20T18:16:25.800353939Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65534, Sequence: (0x80000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:172.18.0.1 "
time="2018-08-20T18:16:25.804956172Z" level=debug msg="/sbin/iptables, [--wait -t filter -n -L DOCKER-USER]"
time="2018-08-20T18:16:25.806482725Z" level=debug msg="/sbin/iptables, [--wait -t filter -C DOCKER-USER -j RETURN]"
time="2018-08-20T18:16:25.808102790Z" level=debug msg="/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.810832595Z" level=debug msg="/sbin/iptables, [--wait -D FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.814517865Z" level=debug msg="/sbin/iptables, [--wait -I FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:25.826014064Z" level=info msg="Loading containers: done."
time="2018-08-20T18:16:25.857361055Z" level=info msg="Docker daemon" commit=e65ae8273f graphdriver(s)=aufs version=dev
time="2018-08-20T18:16:25.857602670Z" level=info msg="Daemon has completed initialization"
time="2018-08-20T18:16:25.864775293Z" level=debug msg="Registering routers"
time="2018-08-20T18:16:25.864811390Z" level=debug msg="Registering GET, /containers/{name:.*}/checkpoints"
time="2018-08-20T18:16:25.864988498Z" level=debug msg="Registering POST, /containers/{name:.*}/checkpoints"
time="2018-08-20T18:16:25.865122945Z" level=debug msg="Registering DELETE, /containers/{name}/checkpoints/{checkpoint}"
time="2018-08-20T18:16:25.865312355Z" level=debug msg="Registering HEAD, /containers/{name:.*}/archive"
time="2018-08-20T18:16:25.865460302Z" level=debug msg="Registering GET, /containers/json"
time="2018-08-20T18:16:25.865556225Z" level=debug msg="Registering GET, /containers/{name:.*}/export"
time="2018-08-20T18:16:25.865695504Z" level=debug msg="Registering GET, /containers/{name:.*}/changes"
time="2018-08-20T18:16:25.865854523Z" level=debug msg="Registering GET, /containers/{name:.*}/json"
time="2018-08-20T18:16:25.865970964Z" level=debug msg="Registering GET, /containers/{name:.*}/top"
time="2018-08-20T18:16:25.866103309Z" level=debug msg="Registering GET, /containers/{name:.*}/logs"
time="2018-08-20T18:16:25.866232837Z" level=debug msg="Registering GET, /containers/{name:.*}/stats"
time="2018-08-20T18:16:25.866348971Z" level=debug msg="Registering GET, /containers/{name:.*}/attach/ws"
time="2018-08-20T18:16:25.866472073Z" level=debug msg="Registering GET, /exec/{id:.*}/json"
time="2018-08-20T18:16:25.866598265Z" level=debug msg="Registering GET, /containers/{name:.*}/archive"
time="2018-08-20T18:16:25.866725199Z" level=debug msg="Registering POST, /containers/create"
time="2018-08-20T18:16:25.866818416Z" level=debug msg="Registering POST, /containers/{name:.*}/kill"
time="2018-08-20T18:16:25.866938188Z" level=debug msg="Registering POST, /containers/{name:.*}/pause"
time="2018-08-20T18:16:25.867066802Z" level=debug msg="Registering POST, /containers/{name:.*}/unpause"
time="2018-08-20T18:16:25.867194858Z" level=debug msg="Registering POST, /containers/{name:.*}/restart"
time="2018-08-20T18:16:25.867319306Z" level=debug msg="Registering POST, /containers/{name:.*}/start"
time="2018-08-20T18:16:25.867469746Z" level=debug msg="Registering POST, /containers/{name:.*}/stop"
time="2018-08-20T18:16:25.867591311Z" level=debug msg="Registering POST, /containers/{name:.*}/wait"
time="2018-08-20T18:16:25.867714829Z" level=debug msg="Registering POST, /containers/{name:.*}/resize"
time="2018-08-20T18:16:25.867837558Z" level=debug msg="Registering POST, /containers/{name:.*}/attach"
time="2018-08-20T18:16:25.867995134Z" level=debug msg="Registering POST, /containers/{name:.*}/copy"
time="2018-08-20T18:16:25.868114782Z" level=debug msg="Registering POST, /containers/{name:.*}/exec"
time="2018-08-20T18:16:25.868234640Z" level=debug msg="Registering POST, /exec/{name:.*}/start"
time="2018-08-20T18:16:25.868340845Z" level=debug msg="Registering POST, /exec/{name:.*}/resize"
time="2018-08-20T18:16:25.868447844Z" level=debug msg="Registering POST, /containers/{name:.*}/rename"
time="2018-08-20T18:16:25.868562892Z" level=debug msg="Registering POST, /containers/{name:.*}/update"
time="2018-08-20T18:16:25.868684656Z" level=debug msg="Registering POST, /containers/prune"
time="2018-08-20T18:16:25.868778300Z" level=debug msg="Registering POST, /commit"
time="2018-08-20T18:16:25.868866278Z" level=debug msg="Registering PUT, /containers/{name:.*}/archive"
time="2018-08-20T18:16:25.869016311Z" level=debug msg="Registering DELETE, /containers/{name:.*}"
time="2018-08-20T18:16:25.869157488Z" level=debug msg="Registering GET, /images/json"
time="2018-08-20T18:16:25.869243856Z" level=debug msg="Registering GET, /images/search"
time="2018-08-20T18:16:25.869339478Z" level=debug msg="Registering GET, /images/get"
time="2018-08-20T18:16:25.869455561Z" level=debug msg="Registering GET, /images/{name:.*}/get"
time="2018-08-20T18:16:25.869577458Z" level=debug msg="Registering GET, /images/{name:.*}/history"
time="2018-08-20T18:16:25.869695129Z" level=debug msg="Registering GET, /images/{name:.*}/json"
time="2018-08-20T18:16:25.869825618Z" level=debug msg="Registering POST, /images/load"
time="2018-08-20T18:16:25.869910010Z" level=debug msg="Registering POST, /images/create"
time="2018-08-20T18:16:25.869989877Z" level=debug msg="Registering POST, /images/{name:.*}/push"
time="2018-08-20T18:16:25.870107105Z" level=debug msg="Registering POST, /images/{name:.*}/tag"
time="2018-08-20T18:16:25.870206983Z" level=debug msg="Registering POST, /images/prune"
time="2018-08-20T18:16:25.870288550Z" level=debug msg="Registering DELETE, /images/{name:.*}"
time="2018-08-20T18:16:25.870397985Z" level=debug msg="Registering OPTIONS, /{anyroute:.*}"
time="2018-08-20T18:16:25.870471745Z" level=debug msg="Registering GET, /_ping"
time="2018-08-20T18:16:25.870521576Z" level=debug msg="Registering GET, /events"
time="2018-08-20T18:16:25.870568688Z" level=debug msg="Registering GET, /info"
time="2018-08-20T18:16:25.870615260Z" level=debug msg="Registering GET, /version"
time="2018-08-20T18:16:25.870658015Z" level=debug msg="Registering GET, /system/df"
time="2018-08-20T18:16:25.870740686Z" level=debug msg="Registering POST, /auth"
time="2018-08-20T18:16:25.870794233Z" level=debug msg="Registering GET, /volumes"
time="2018-08-20T18:16:25.870851255Z" level=debug msg="Registering GET, /volumes/{name:.*}"
time="2018-08-20T18:16:25.870932372Z" level=debug msg="Registering POST, /volumes/create"
time="2018-08-20T18:16:25.870995579Z" level=debug msg="Registering POST, /volumes/prune"
time="2018-08-20T18:16:25.871047911Z" level=debug msg="Registering DELETE, /volumes/{name:.*}"
time="2018-08-20T18:16:25.871123453Z" level=debug msg="Registering POST, /build"
time="2018-08-20T18:16:25.871165564Z" level=debug msg="Registering POST, /build/prune"
time="2018-08-20T18:16:25.871214245Z" level=debug msg="Registering POST, /build/cancel"
time="2018-08-20T18:16:25.871267310Z" level=debug msg="Registering POST, /session"
time="2018-08-20T18:16:25.871316360Z" level=debug msg="Registering POST, /swarm/init"
time="2018-08-20T18:16:25.871367369Z" level=debug msg="Registering POST, /swarm/join"
time="2018-08-20T18:16:25.871413186Z" level=debug msg="Registering POST, /swarm/leave"
time="2018-08-20T18:16:25.871463200Z" level=debug msg="Registering GET, /swarm"
time="2018-08-20T18:16:25.871507390Z" level=debug msg="Registering GET, /swarm/unlockkey"
time="2018-08-20T18:16:25.871555565Z" level=debug msg="Registering POST, /swarm/update"
time="2018-08-20T18:16:25.871602777Z" level=debug msg="Registering POST, /swarm/unlock"
time="2018-08-20T18:16:25.871655752Z" level=debug msg="Registering GET, /services"
time="2018-08-20T18:16:25.871706811Z" level=debug msg="Registering GET, /services/{id}"
time="2018-08-20T18:16:25.871770380Z" level=debug msg="Registering POST, /services/create"
time="2018-08-20T18:16:25.871821621Z" level=debug msg="Registering POST, /services/{id}/update"
time="2018-08-20T18:16:25.872104089Z" level=debug msg="Registering DELETE, /services/{id}"
time="2018-08-20T18:16:25.872310734Z" level=debug msg="Registering GET, /services/{id}/logs"
time="2018-08-20T18:16:25.872655343Z" level=debug msg="Registering GET, /nodes"
time="2018-08-20T18:16:25.872966973Z" level=debug msg="Registering GET, /nodes/{id}"
time="2018-08-20T18:16:25.873467763Z" level=debug msg="Registering DELETE, /nodes/{id}"
time="2018-08-20T18:16:25.873635302Z" level=debug msg="Registering POST, /nodes/{id}/update"
time="2018-08-20T18:16:25.873849107Z" level=debug msg="Registering GET, /tasks"
time="2018-08-20T18:16:25.874014139Z" level=debug msg="Registering GET, /tasks/{id}"
time="2018-08-20T18:16:25.874174092Z" level=debug msg="Registering GET, /tasks/{id}/logs"
time="2018-08-20T18:16:25.874527577Z" level=debug msg="Registering GET, /secrets"
time="2018-08-20T18:16:25.874680591Z" level=debug msg="Registering POST, /secrets/create"
time="2018-08-20T18:16:25.874877394Z" level=debug msg="Registering DELETE, /secrets/{id}"
time="2018-08-20T18:16:25.875110248Z" level=debug msg="Registering GET, /secrets/{id}"
time="2018-08-20T18:16:25.875273577Z" level=debug msg="Registering POST, /secrets/{id}/update"
time="2018-08-20T18:16:25.875519752Z" level=debug msg="Registering GET, /configs"
time="2018-08-20T18:16:25.875786917Z" level=debug msg="Registering POST, /configs/create"
time="2018-08-20T18:16:25.875988799Z" level=debug msg="Registering DELETE, /configs/{id}"
time="2018-08-20T18:16:25.876139046Z" level=debug msg="Registering GET, /configs/{id}"
time="2018-08-20T18:16:25.876346472Z" level=debug msg="Registering POST, /configs/{id}/update"
time="2018-08-20T18:16:25.876582439Z" level=debug msg="Registering GET, /plugins"
time="2018-08-20T18:16:25.876761418Z" level=debug msg="Registering GET, /plugins/{name:.*}/json"
time="2018-08-20T18:16:25.876959252Z" level=debug msg="Registering GET, /plugins/privileges"
time="2018-08-20T18:16:25.877215129Z" level=debug msg="Registering DELETE, /plugins/{name:.*}"
time="2018-08-20T18:16:25.877817509Z" level=debug msg="Registering POST, /plugins/{name:.*}/enable"
time="2018-08-20T18:16:25.877944694Z" level=debug msg="Registering POST, /plugins/{name:.*}/disable"
time="2018-08-20T18:16:25.878346564Z" level=debug msg="Registering POST, /plugins/pull"
time="2018-08-20T18:16:25.878450407Z" level=debug msg="Registering POST, /plugins/{name:.*}/push"
time="2018-08-20T18:16:25.878578843Z" level=debug msg="Registering POST, /plugins/{name:.*}/upgrade"
time="2018-08-20T18:16:25.878717232Z" level=debug msg="Registering POST, /plugins/{name:.*}/set"
time="2018-08-20T18:16:25.878847581Z" level=debug msg="Registering POST, /plugins/create"
time="2018-08-20T18:16:25.878938752Z" level=debug msg="Registering GET, /distribution/{name:.*}/json"
time="2018-08-20T18:16:25.879094789Z" level=debug msg="Registering GET, /networks"
time="2018-08-20T18:16:25.879192719Z" level=debug msg="Registering GET, /networks/"
time="2018-08-20T18:16:25.879281722Z" level=debug msg="Registering GET, /networks/{id:.+}"
time="2018-08-20T18:16:25.879430346Z" level=debug msg="Registering POST, /networks/create"
time="2018-08-20T18:16:25.879540913Z" level=debug msg="Registering POST, /networks/{id:.*}/connect"
time="2018-08-20T18:16:25.879675774Z" level=debug msg="Registering POST, /networks/{id:.*}/disconnect"
time="2018-08-20T18:16:25.879841269Z" level=debug msg="Registering POST, /networks/prune"
time="2018-08-20T18:16:25.879974488Z" level=debug msg="Registering DELETE, /networks/{id:.*}"
time="2018-08-20T18:16:25.880606103Z" level=info msg="API listen on /tmp/docker-integration/d3212059849e4.sock"
time="2018-08-20T18:16:26.192484587Z" level=debug msg="Calling GET /_ping"
time="2018-08-20T18:16:26.193260053Z" level=debug msg="Calling GET /info"
time="2018-08-20T18:16:26.244273797Z" level=debug msg="Calling POST /v1.39/images/load?quiet=1"
time="2018-08-20T18:16:26.432295045Z" level=debug msg="Applied tar sha256:0271b8eebde3fa9a6126b1f2335e170f902731ab4942f9f1914e77016540c7bb to dd8942717ce89a57f397d65d5c6d29c01b6761371c2fe2f3cf17446e6152f2ee, size: 1129289"
time="2018-08-20T18:16:26.441655128Z" level=debug msg="Calling POST /v1.39/containers/create"
time="2018-08-20T18:16:26.442219165Z" level=debug msg="form data: {\"AttachStderr\":false,\"AttachStdin\":false,\"AttachStdout\":false,\"Cmd\":[\"top\"],\"Domainname\":\"\",\"Entrypoint\":null,\"Env\":null,\"HostConfig\":{\"AutoRemove\":false,\"Binds\":null,\"BlkioDeviceReadBps\":null,\"BlkioDeviceReadIOps\":null,\"BlkioDeviceWriteBps\":null,\"BlkioDeviceWriteIOps\":null,\"BlkioWeight\":0,\"BlkioWeightDevice\":null,\"CapAdd\":null,\"CapDrop\":null,\"Cgroup\":\"\",\"CgroupParent\":\"\",\"ConsoleSize\":[0,0],\"ContainerIDFile\":\"\",\"CpuCount\":0,\"CpuPercent\":0,\"CpuPeriod\":0,\"CpuQuota\":0,\"CpuRealtimePeriod\":0,\"CpuRealtimeRuntime\":0,\"CpuShares\":0,\"CpusetCpus\":\"\",\"CpusetMems\":\"\",\"DeviceCgroupRules\":null,\"Devices\":null,\"DiskQuota\":0,\"Dns\":null,\"DnsOptions\":null,\"DnsSearch\":null,\"ExtraHosts\":null,\"GroupAdd\":null,\"IOMaximumBandwidth\":0,\"IOMaximumIOps\":0,\"IpcMode\":\"\",\"Isolation\":\"\",\"KernelMemory\":0,\"Links\":null,\"LogConfig\":{\"Config\":null,\"Type\":\"\"},\"MaskedPaths\":null,\"Memory\":0,\"MemoryReservation\":0,\"MemorySwap\":0,\"MemorySwappiness\":null,\"NanoCpus\":0,\"NetworkMode\":\"\",\"OomKillDisable\":null,\"OomScoreAdj\":0,\"PidMode\":\"\",\"PidsLimit\":0,\"PortBindings\":null,\"Privileged\":false,\"PublishAllPorts\":false,\"ReadonlyPaths\":null,\"ReadonlyRootfs\":false,\"RestartPolicy\":{\"MaximumRetryCount\":0,\"Name\":\"\"},\"SecurityOpt\":null,\"ShmSize\":0,\"UTSMode\":\"\",\"Ulimits\":null,\"UsernsMode\":\"\",\"VolumeDriver\":\"\",\"VolumesFrom\":null},\"Hostname\":\"\",\"Image\":\"busybox\",\"Labels\":null,\"NetworkingConfig\":{\"EndpointsConfig\":null},\"OnBuild\":null,\"OpenStdin\":false,\"StdinOnce\":false,\"Tty\":false,\"User\":\"\",\"Volumes\":null,\"WorkingDir\":\"\"}"
time="2018-08-20T18:16:26.479990679Z" level=debug msg="container mounted via layerStore: &{/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7 0x324cd40 0x324cd40}"
time="2018-08-20T18:16:26.500098900Z" level=debug msg="Calling POST /v1.39/containers/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2/start"
time="2018-08-20T18:16:26.501560995Z" level=debug msg="container mounted via layerStore: &{/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7 0x324cd40 0x324cd40}"
time="2018-08-20T18:16:26.502120047Z" level=debug msg="Assigning addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:26.502145578Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, <nil>, map[])"
time="2018-08-20T18:16:26.502196490Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65533, Sequence: (0xc0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:<nil> "
time="2018-08-20T18:16:26.506036366Z" level=debug msg="Assigning addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:26.511047893Z" level=debug msg="Programming external connectivity on endpoint pedantic_knuth (f973c59460352360198e93fe5c43f98bce86c09f86925927980eed33bcc03794)"
time="2018-08-20T18:16:26.511778072Z" level=debug msg="EnableService 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 START"
time="2018-08-20T18:16:26.511799485Z" level=debug msg="EnableService 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 DONE"
time="2018-08-20T18:16:26.514678287Z" level=debug msg="bundle dir created" bundle=/tmp/docker-execroot/d3212059849e4/containerd/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 module=libcontainerd namespace=moby root=/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7
time="2018-08-20T18:16:26.717331473Z" level=debug msg="sandbox set key processing took 82.634737ms for container 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2"
time="2018-08-20T18:16:26.825271060Z" level=debug msg=event module=libcontainerd namespace=moby topic=/tasks/create
time="2018-08-20T18:16:26.845180535Z" level=debug msg=event module=libcontainerd namespace=moby topic=/tasks/start
time="2018-08-20T18:16:26.858966410Z" level=debug msg="Calling GET /v1.39/containers/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2/json"
time="2018-08-20T18:16:26.889937131Z" level=debug msg="Listener created for HTTP on unix (/tmp/docker-integration/d3212059849e4.sock)"
time="2018-08-20T18:16:26.892479453Z" level=debug msg="Golang's threads limit set to 115380"
time="2018-08-20T18:16:26.894049069Z" level=info msg="parsed scheme: \"unix\"" module=grpc
time="2018-08-20T18:16:26.894070102Z" level=info msg="scheme \"unix\" not registered, fallback to default scheme" module=grpc
time="2018-08-20T18:16:26.894146046Z" level=info msg="parsed scheme: \"unix\"" module=grpc
time="2018-08-20T18:16:26.894161467Z" level=info msg="scheme \"unix\" not registered, fallback to default scheme" module=grpc
time="2018-08-20T18:16:26.894303898Z" level=info msg="ccResolverWrapper: sending new addresses to cc: [{unix:///var/run/docker/containerd/docker-containerd.sock 0  <nil>}]" module=grpc
time="2018-08-20T18:16:26.894359729Z" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
time="2018-08-20T18:16:26.894464535Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc420418a90, CONNECTING" module=grpc
time="2018-08-20T18:16:26.894469017Z" level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
time="2018-08-20T18:16:26.894362759Z" level=debug msg="Using default logging driver json-file"
time="2018-08-20T18:16:26.894561555Z" level=debug msg="[graphdriver] trying provided driver: aufs"
time="2018-08-20T18:16:26.894776831Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc420418a90, READY" module=grpc
time="2018-08-20T18:16:26.894838968Z" level=debug msg="processing event stream" module=libcontainerd namespace=plugins.moby
time="2018-08-20T18:16:26.895977060Z" level=debug msg="Initialized graph driver aufs"
time="2018-08-20T18:16:26.896859393Z" level=info msg="ccResolverWrapper: sending new addresses to cc: [{unix:///var/run/docker/containerd/docker-containerd.sock 0  <nil>}]" module=grpc
time="2018-08-20T18:16:26.897475981Z" level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
time="2018-08-20T18:16:26.897893889Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc4201963f0, CONNECTING" module=grpc
time="2018-08-20T18:16:26.899157419Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
time="2018-08-20T18:16:26.899117668Z" level=info msg="pickfirstBalancer: HandleSubConnStateChange: 0xc4201963f0, READY" module=grpc
time="2018-08-20T18:16:26.899508166Z" level=warning msg="Your kernel does not support swap memory limit"
time="2018-08-20T18:16:26.899588283Z" level=warning msg="Your kernel does not support cgroup rt period"
time="2018-08-20T18:16:26.899613605Z" level=warning msg="Your kernel does not support cgroup rt runtime"
time="2018-08-20T18:16:26.900257742Z" level=debug msg="Max Concurrent Downloads: 3"
time="2018-08-20T18:16:26.900283613Z" level=debug msg="Max Concurrent Uploads: 5"
time="2018-08-20T18:16:26.900318584Z" level=info msg="Loading containers: start."
time="2018-08-20T18:16:26.900740415Z" level=debug msg="processing event stream" module=libcontainerd namespace=moby
time="2018-08-20T18:16:26.901602399Z" level=debug msg="Loaded container 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2, isRunning: true"
time="2018-08-20T18:16:26.907872874Z" level=debug msg="restoring container" container=89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 paused=false running=true
time="2018-08-20T18:16:26.909655423Z" level=debug msg="restored container" alive=false container=89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 module=libcontainerd namespace=moby pid=0
time="2018-08-20T18:16:26.973449048Z" level=debug msg="container mounted via layerStore: &{/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7 0x324cd40 0x324cd40}"
time="2018-08-20T18:16:26.988837103Z" level=debug msg="Option Experimental: false"
time="2018-08-20T18:16:26.988892099Z" level=debug msg="Option DefaultDriver: bridge"
time="2018-08-20T18:16:26.988910663Z" level=debug msg="Option DefaultNetwork: bridge"
time="2018-08-20T18:16:26.988926977Z" level=debug msg="Network Control Plane MTU: 1500"
time="2018-08-20T18:16:26.990216911Z" level=debug msg="Network (2836ffb) restored"
time="2018-08-20T18:16:26.990602428Z" level=debug msg="Endpoint (f973c59) restored to network (2836ffb)"
time="2018-08-20T18:16:27.016523528Z" level=debug msg="Allocating IPv4 pools for network bridge (2836ffbe26dad9e6561c82801dce7c4d42067335063075ff77bbf7e3a6db7d25)"
time="2018-08-20T18:16:27.016577384Z" level=debug msg="RequestPool(LocalDefault, 172.18.0.0/16, , map[], false)"
time="2018-08-20T18:16:27.016635030Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, 172.18.0.1, map[RequestAddressType:com.docker.network.gateway])"
time="2018-08-20T18:16:27.016722790Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65534, Sequence: (0x80000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:172.18.0.1 "
time="2018-08-20T18:16:27.017148119Z" level=debug msg="Assigning addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:27.017179889Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, 172.18.0.2, map[])"
time="2018-08-20T18:16:27.017211382Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65533, Sequence: (0xc0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:172.18.0.2 "
time="2018-08-20T18:16:27.304443358Z" level=info msg="Removing stale sandbox 2450291bac53de42734983c99c05e7bf21d6b09483660f899a2f70577f39fb8b (89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2)"
time="2018-08-20T18:16:27.304552676Z" level=debug msg="Revoking external connectivity on endpoint pedantic_knuth (f973c59460352360198e93fe5c43f98bce86c09f86925927980eed33bcc03794)"
time="2018-08-20T18:16:27.306398885Z" level=debug msg="DeleteConntrackEntries purged ipv4:0, ipv6:0"
time="2018-08-20T18:16:27.308898736Z" level=warning msg="Error (Unable to complete atomic operation, key modified) deleting object [endpoint 2836ffbe26dad9e6561c82801dce7c4d42067335063075ff77bbf7e3a6db7d25 f973c59460352360198e93fe5c43f98bce86c09f86925927980eed33bcc03794], retrying...."
time="2018-08-20T18:16:27.310406872Z" level=debug msg="Releasing addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:27.310458673Z" level=debug msg="ReleaseAddress(LocalDefault/172.18.0.0/16, 172.18.0.2)"
time="2018-08-20T18:16:27.310510563Z" level=debug msg="Released address PoolID:LocalDefault/172.18.0.0/16, Address:172.18.0.2 Sequence:App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65532, Sequence: (0xe0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0"
time="2018-08-20T18:16:27.325575221Z" level=debug msg="releasing IPv4 pools from network bridge (2836ffbe26dad9e6561c82801dce7c4d42067335063075ff77bbf7e3a6db7d25)"
time="2018-08-20T18:16:27.325628573Z" level=debug msg="ReleaseAddress(LocalDefault/172.18.0.0/16, 172.18.0.1)"
time="2018-08-20T18:16:27.325678600Z" level=debug msg="Released address PoolID:LocalDefault/172.18.0.0/16, Address:172.18.0.1 Sequence:App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65533, Sequence: (0xc0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0"
time="2018-08-20T18:16:27.325713125Z" level=debug msg="ReleasePool(LocalDefault/172.18.0.0/16)"
time="2018-08-20T18:16:27.326592406Z" level=debug msg="cleanupServiceDiscovery for network:2836ffbe26dad9e6561c82801dce7c4d42067335063075ff77bbf7e3a6db7d25"
time="2018-08-20T18:16:27.328300373Z" level=info msg="Default bridge (docker0) is assigned with an IP address 172.18.0.0/16. Daemon option --bip can be used to set a preferred IP address"
time="2018-08-20T18:16:27.328360937Z" level=debug msg="Allocating IPv4 pools for network bridge (89d549b29258edac5be3cba9c0d226e7ef09dbfa86bc0643411f2528fd76ec67)"
time="2018-08-20T18:16:27.328391553Z" level=debug msg="RequestPool(LocalDefault, 172.18.0.0/16, , map[], false)"
time="2018-08-20T18:16:27.328508708Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, 172.18.0.1, map[RequestAddressType:com.docker.network.gateway])"
time="2018-08-20T18:16:27.328556076Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65534, Sequence: (0x80000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:172.18.0.1 "
time="2018-08-20T18:16:27.336951996Z" level=warning msg="Running modprobe nf_nat failed with message: ``, error: exec: \"modprobe\": executable file not found in $PATH"
time="2018-08-20T18:16:27.337051905Z" level=warning msg="Running modprobe xt_conntrack failed with message: ``, error: exec: \"modprobe\": executable file not found in $PATH"
time="2018-08-20T18:16:27.337141538Z" level=debug msg="Fail to initialize firewalld: Failed to connect to D-Bus system bus: dial unix /var/run/dbus/system_bus_socket: connect: no such file or directory, using raw iptables instead"
time="2018-08-20T18:16:27.340874458Z" level=debug msg="/sbin/iptables, [--wait -t filter -n -L DOCKER-USER]"
time="2018-08-20T18:16:27.342619950Z" level=debug msg="/sbin/iptables, [--wait -t filter -C DOCKER-USER -j RETURN]"
time="2018-08-20T18:16:27.344232796Z" level=debug msg="/sbin/iptables, [--wait -t filter -C FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:27.346367615Z" level=debug msg="/sbin/iptables, [--wait -D FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:27.348182290Z" level=debug msg="/sbin/iptables, [--wait -I FORWARD -j DOCKER-USER]"
time="2018-08-20T18:16:27.351263227Z" level=info msg="Loading containers: done."
time="2018-08-20T18:16:27.358963642Z" level=info msg="Docker daemon" commit=e65ae8273f graphdriver(s)=aufs version=dev
time="2018-08-20T18:16:27.359077875Z" level=info msg="Daemon has completed initialization"
time="2018-08-20T18:16:27.362772266Z" level=debug msg="Registering routers"
time="2018-08-20T18:16:27.362796348Z" level=debug msg="Registering GET, /containers/{name:.*}/checkpoints"
time="2018-08-20T18:16:27.362923728Z" level=debug msg="Registering POST, /containers/{name:.*}/checkpoints"
time="2018-08-20T18:16:27.363016331Z" level=debug msg="Registering DELETE, /containers/{name}/checkpoints/{checkpoint}"
time="2018-08-20T18:16:27.363137471Z" level=debug msg="Registering HEAD, /containers/{name:.*}/archive"
time="2018-08-20T18:16:27.363221624Z" level=debug msg="Registering GET, /containers/json"
time="2018-08-20T18:16:27.363276450Z" level=debug msg="Registering GET, /containers/{name:.*}/export"
time="2018-08-20T18:16:27.363358432Z" level=debug msg="Registering GET, /containers/{name:.*}/changes"
time="2018-08-20T18:16:27.363457983Z" level=debug msg="Registering GET, /containers/{name:.*}/json"
time="2018-08-20T18:16:27.363557042Z" level=debug msg="Registering GET, /containers/{name:.*}/top"
time="2018-08-20T18:16:27.363653029Z" level=debug msg="Registering GET, /containers/{name:.*}/logs"
time="2018-08-20T18:16:27.363730833Z" level=debug msg="Registering GET, /containers/{name:.*}/stats"
time="2018-08-20T18:16:27.363803974Z" level=debug msg="Registering GET, /containers/{name:.*}/attach/ws"
time="2018-08-20T18:16:27.363903577Z" level=debug msg="Registering GET, /exec/{id:.*}/json"
time="2018-08-20T18:16:27.364006551Z" level=debug msg="Registering GET, /containers/{name:.*}/archive"
time="2018-08-20T18:16:27.364096927Z" level=debug msg="Registering POST, /containers/create"
time="2018-08-20T18:16:27.364178486Z" level=debug msg="Registering POST, /containers/{name:.*}/kill"
time="2018-08-20T18:16:27.364262881Z" level=debug msg="Registering POST, /containers/{name:.*}/pause"
time="2018-08-20T18:16:27.364333827Z" level=debug msg="Registering POST, /containers/{name:.*}/unpause"
time="2018-08-20T18:16:27.364409490Z" level=debug msg="Registering POST, /containers/{name:.*}/restart"
time="2018-08-20T18:16:27.364483883Z" level=debug msg="Registering POST, /containers/{name:.*}/start"
time="2018-08-20T18:16:27.364561390Z" level=debug msg="Registering POST, /containers/{name:.*}/stop"
time="2018-08-20T18:16:27.364632987Z" level=debug msg="Registering POST, /containers/{name:.*}/wait"
time="2018-08-20T18:16:27.364705807Z" level=debug msg="Registering POST, /containers/{name:.*}/resize"
time="2018-08-20T18:16:27.364773769Z" level=debug msg="Registering POST, /containers/{name:.*}/attach"
time="2018-08-20T18:16:27.364852317Z" level=debug msg="Registering POST, /containers/{name:.*}/copy"
time="2018-08-20T18:16:27.364947108Z" level=debug msg="Registering POST, /containers/{name:.*}/exec"
time="2018-08-20T18:16:27.365021870Z" level=debug msg="Registering POST, /exec/{name:.*}/start"
time="2018-08-20T18:16:27.365091125Z" level=debug msg="Registering POST, /exec/{name:.*}/resize"
time="2018-08-20T18:16:27.365159258Z" level=debug msg="Registering POST, /containers/{name:.*}/rename"
time="2018-08-20T18:16:27.365233555Z" level=debug msg="Registering POST, /containers/{name:.*}/update"
time="2018-08-20T18:16:27.365306797Z" level=debug msg="Registering POST, /containers/prune"
time="2018-08-20T18:16:27.365359485Z" level=debug msg="Registering POST, /commit"
time="2018-08-20T18:16:27.365405227Z" level=debug msg="Registering PUT, /containers/{name:.*}/archive"
time="2018-08-20T18:16:27.365478959Z" level=debug msg="Registering DELETE, /containers/{name:.*}"
time="2018-08-20T18:16:27.365549770Z" level=debug msg="Registering GET, /images/json"
time="2018-08-20T18:16:27.365602437Z" level=debug msg="Registering GET, /images/search"
time="2018-08-20T18:16:27.365693434Z" level=debug msg="Registering GET, /images/get"
time="2018-08-20T18:16:27.365840103Z" level=debug msg="Registering GET, /images/{name:.*}/get"
time="2018-08-20T18:16:27.365959880Z" level=debug msg="Registering GET, /images/{name:.*}/history"
time="2018-08-20T18:16:27.366081871Z" level=debug msg="Registering GET, /images/{name:.*}/json"
time="2018-08-20T18:16:27.366180397Z" level=debug msg="Registering POST, /images/load"
time="2018-08-20T18:16:27.366258552Z" level=debug msg="Registering POST, /images/create"
time="2018-08-20T18:16:27.366321643Z" level=debug msg="Registering POST, /images/{name:.*}/push"
time="2018-08-20T18:16:27.366421373Z" level=debug msg="Registering POST, /images/{name:.*}/tag"
time="2018-08-20T18:16:27.366540426Z" level=debug msg="Registering POST, /images/prune"
time="2018-08-20T18:16:27.366599687Z" level=debug msg="Registering DELETE, /images/{name:.*}"
time="2018-08-20T18:16:27.366663402Z" level=debug msg="Registering OPTIONS, /{anyroute:.*}"
time="2018-08-20T18:16:27.366732309Z" level=debug msg="Registering GET, /_ping"
time="2018-08-20T18:16:27.366778064Z" level=debug msg="Registering GET, /events"
time="2018-08-20T18:16:27.366823783Z" level=debug msg="Registering GET, /info"
time="2018-08-20T18:16:27.366865696Z" level=debug msg="Registering GET, /version"
time="2018-08-20T18:16:27.366926623Z" level=debug msg="Registering GET, /system/df"
time="2018-08-20T18:16:27.367004529Z" level=debug msg="Registering POST, /auth"
time="2018-08-20T18:16:27.367048578Z" level=debug msg="Registering GET, /volumes"
time="2018-08-20T18:16:27.367106900Z" level=debug msg="Registering GET, /volumes/{name:.*}"
time="2018-08-20T18:16:27.367179212Z" level=debug msg="Registering POST, /volumes/create"
time="2018-08-20T18:16:27.367230953Z" level=debug msg="Registering POST, /volumes/prune"
time="2018-08-20T18:16:27.367289485Z" level=debug msg="Registering DELETE, /volumes/{name:.*}"
time="2018-08-20T18:16:27.367371085Z" level=debug msg="Registering POST, /build"
time="2018-08-20T18:16:27.367417379Z" level=debug msg="Registering POST, /build/prune"
time="2018-08-20T18:16:27.367470344Z" level=debug msg="Registering POST, /build/cancel"
time="2018-08-20T18:16:27.367531113Z" level=debug msg="Registering POST, /session"
time="2018-08-20T18:16:27.367586973Z" level=debug msg="Registering POST, /swarm/init"
time="2018-08-20T18:16:27.367648226Z" level=debug msg="Registering POST, /swarm/join"
time="2018-08-20T18:16:27.367724488Z" level=debug msg="Registering POST, /swarm/leave"
time="2018-08-20T18:16:27.367808711Z" level=debug msg="Registering GET, /swarm"
time="2018-08-20T18:16:27.367919320Z" level=debug msg="Registering GET, /swarm/unlockkey"
time="2018-08-20T18:16:27.367985962Z" level=debug msg="Registering POST, /swarm/update"
time="2018-08-20T18:16:27.368044605Z" level=debug msg="Registering POST, /swarm/unlock"
time="2018-08-20T18:16:27.368104483Z" level=debug msg="Registering GET, /services"
time="2018-08-20T18:16:27.368146657Z" level=debug msg="Registering GET, /services/{id}"
time="2018-08-20T18:16:27.368223561Z" level=debug msg="Registering POST, /services/create"
time="2018-08-20T18:16:27.368277960Z" level=debug msg="Registering POST, /services/{id}/update"
time="2018-08-20T18:16:27.368363892Z" level=debug msg="Registering DELETE, /services/{id}"
time="2018-08-20T18:16:27.368451970Z" level=debug msg="Registering GET, /services/{id}/logs"
time="2018-08-20T18:16:27.368537915Z" level=debug msg="Registering GET, /nodes"
time="2018-08-20T18:16:27.368577068Z" level=debug msg="Registering GET, /nodes/{id}"
time="2018-08-20T18:16:27.368657142Z" level=debug msg="Registering DELETE, /nodes/{id}"
time="2018-08-20T18:16:27.368731311Z" level=debug msg="Registering POST, /nodes/{id}/update"
time="2018-08-20T18:16:27.368815143Z" level=debug msg="Registering GET, /tasks"
time="2018-08-20T18:16:27.368861994Z" level=debug msg="Registering GET, /tasks/{id}"
time="2018-08-20T18:16:27.368941002Z" level=debug msg="Registering GET, /tasks/{id}/logs"
time="2018-08-20T18:16:27.369026047Z" level=debug msg="Registering GET, /secrets"
time="2018-08-20T18:16:27.369094254Z" level=debug msg="Registering POST, /secrets/create"
time="2018-08-20T18:16:27.369146669Z" level=debug msg="Registering DELETE, /secrets/{id}"
time="2018-08-20T18:16:27.369231966Z" level=debug msg="Registering GET, /secrets/{id}"
time="2018-08-20T18:16:27.369289891Z" level=debug msg="Registering POST, /secrets/{id}/update"
time="2018-08-20T18:16:27.369403546Z" level=debug msg="Registering GET, /configs"
time="2018-08-20T18:16:27.369464780Z" level=debug msg="Registering POST, /configs/create"
time="2018-08-20T18:16:27.369524131Z" level=debug msg="Registering DELETE, /configs/{id}"
time="2018-08-20T18:16:27.369590061Z" level=debug msg="Registering GET, /configs/{id}"
time="2018-08-20T18:16:27.369668340Z" level=debug msg="Registering POST, /configs/{id}/update"
time="2018-08-20T18:16:27.369812921Z" level=debug msg="Registering GET, /plugins"
time="2018-08-20T18:16:27.369880690Z" level=debug msg="Registering GET, /plugins/{name:.*}/json"
time="2018-08-20T18:16:27.369981885Z" level=debug msg="Registering GET, /plugins/privileges"
time="2018-08-20T18:16:27.370044292Z" level=debug msg="Registering DELETE, /plugins/{name:.*}"
time="2018-08-20T18:16:27.370128935Z" level=debug msg="Registering POST, /plugins/{name:.*}/enable"
time="2018-08-20T18:16:27.370206090Z" level=debug msg="Registering POST, /plugins/{name:.*}/disable"
time="2018-08-20T18:16:27.370280482Z" level=debug msg="Registering POST, /plugins/pull"
time="2018-08-20T18:16:27.370339918Z" level=debug msg="Registering POST, /plugins/{name:.*}/push"
time="2018-08-20T18:16:27.370413501Z" level=debug msg="Registering POST, /plugins/{name:.*}/upgrade"
time="2018-08-20T18:16:27.370507482Z" level=debug msg="Registering POST, /plugins/{name:.*}/set"
time="2018-08-20T18:16:27.370660310Z" level=debug msg="Registering POST, /plugins/create"
time="2018-08-20T18:16:27.370765048Z" level=debug msg="Registering GET, /distribution/{name:.*}/json"
time="2018-08-20T18:16:27.370851410Z" level=debug msg="Registering GET, /networks"
time="2018-08-20T18:16:27.370908179Z" level=debug msg="Registering GET, /networks/"
time="2018-08-20T18:16:27.370963427Z" level=debug msg="Registering GET, /networks/{id:.+}"
time="2018-08-20T18:16:27.371042134Z" level=debug msg="Registering POST, /networks/create"
time="2018-08-20T18:16:27.371101552Z" level=debug msg="Registering POST, /networks/{id:.*}/connect"
time="2018-08-20T18:16:27.371183212Z" level=debug msg="Registering POST, /networks/{id:.*}/disconnect"
time="2018-08-20T18:16:27.371328047Z" level=debug msg="Registering POST, /networks/prune"
time="2018-08-20T18:16:27.371434283Z" level=debug msg="Registering DELETE, /networks/{id:.*}"
time="2018-08-20T18:16:27.372104917Z" level=info msg="API listen on /tmp/docker-integration/d3212059849e4.sock"
time="2018-08-20T18:16:27.377867258Z" level=debug msg="Calling GET /_ping"
time="2018-08-20T18:16:27.378726345Z" level=debug msg="Calling GET /info"
time="2018-08-20T18:16:27.390621898Z" level=debug msg="Calling POST /v1.39/containers/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2/start"
time="2018-08-20T18:16:27.391998870Z" level=debug msg="container mounted via layerStore: &{/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7 0x324cd40 0x324cd40}"
time="2018-08-20T18:16:27.392491416Z" level=debug msg="Assigning addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:27.392524950Z" level=debug msg="RequestAddress(LocalDefault/172.18.0.0/16, <nil>, map[])"
time="2018-08-20T18:16:27.392568756Z" level=debug msg="Request address PoolID:172.18.0.0/16 App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65533, Sequence: (0xc0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:0 Serial:false PrefAddress:<nil> "
time="2018-08-20T18:16:27.395302958Z" level=debug msg="Assigning addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:27.399280449Z" level=debug msg="Programming external connectivity on endpoint pedantic_knuth (0956c3f8dad16db4d9b6201c5c4349bca8cf758e736ec4f3d57aeb5059ce970b)"
time="2018-08-20T18:16:27.399996071Z" level=debug msg="EnableService 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 START"
time="2018-08-20T18:16:27.400028620Z" level=debug msg="EnableService 89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 DONE"
time="2018-08-20T18:16:27.402609063Z" level=debug msg="bundle dir created" bundle=/tmp/docker-execroot/d3212059849e4/containerd/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 module=libcontainerd namespace=moby root=/go/src/github.com/docker/docker/bundles/test-integration/d3212059849e4/root/aufs/mnt/96dfcaa9e71b57c7f43a916cd01282152a243642ee985ce8c1d7c47307a017d7
time="2018-08-20T18:16:27.415506103Z" level=error msg="stream copy error: reading from a closed fifo"
time="2018-08-20T18:16:27.415547883Z" level=error msg="stream copy error: reading from a closed fifo"
time="2018-08-20T18:16:27.420682044Z" level=debug msg="Revoking external connectivity on endpoint pedantic_knuth (0956c3f8dad16db4d9b6201c5c4349bca8cf758e736ec4f3d57aeb5059ce970b)"
time="2018-08-20T18:16:27.422646166Z" level=debug msg="DeleteConntrackEntries purged ipv4:0, ipv6:0"
time="2018-08-20T18:16:27.465052311Z" level=debug msg="Releasing addresses for endpoint pedantic_knuth's interface on network bridge"
time="2018-08-20T18:16:27.465100626Z" level=debug msg="ReleaseAddress(LocalDefault/172.18.0.0/16, 172.18.0.2)"
time="2018-08-20T18:16:27.465146784Z" level=debug msg="Released address PoolID:LocalDefault/172.18.0.0/16, Address:172.18.0.2 Sequence:App: ipam/default/data, ID: LocalDefault/172.18.0.0/16, DBIndex: 0x0, Bits: 65536, Unselected: 65532, Sequence: (0xe0000000, 1)->(0x0, 2046)->(0x1, 1)->end Curr:3"
time="2018-08-20T18:16:27.501110818Z" level=error msg="89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2 cleanup: failed to delete container from containerd: no such container"
time="2018-08-20T18:16:27.501179019Z" level=error msg="Handler for POST /v1.39/containers/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2/start returned error: mkdir /var/run/docker/containerd/daemon/io.containerd.runtime.v1.linux/moby/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2: file exists: already exists"
time="2018-08-20T18:16:27.502314513Z" level=debug msg="Calling DELETE /v1.39/containers/89573952a64357960bddc8f55722856e30a3f42e1cdf7ac401560535df7b9bc2?force=1"
time="2018-08-20T18:16:27.509717234Z" level=info msg="Processing signal 'interrupt'"
time="2018-08-20T18:16:27.509878553Z" level=debug msg="daemon configured with a 15 seconds minimum shutdown timeout"
time="2018-08-20T18:16:27.509925544Z" level=debug msg="start clean shutdown of all containers with a 15 seconds timeout..."
time="2018-08-20T18:16:27.510931666Z" level=debug msg="Unix socket /run/docker/libnetwork/4f603034c974a82c514d5bba7f427b1d270302e86e362dd1d1accbe6d0c8509d.sock doesn't exist. cannot accept client connections"
time="2018-08-20T18:16:27.511010726Z" level=debug msg="Cleaning up old mountid : start."
time="2018-08-20T18:16:27.511046930Z" level=info msg="stopping event stream following graceful shutdown" error="<nil>" module=libcontainerd namespace=moby
time="2018-08-20T18:16:27.511621835Z" level=debug msg="Cleaning up old mountid : done."
time="2018-08-20T18:16:27.511949145Z" level=debug msg="Clean shutdown succeeded"
time="2018-08-20T18:16:27.512040345Z" level=info msg="stopping event stream following graceful shutdown" error="context canceled" module=libcontainerd namespace=plugins.moby
```

cc @crosbymichael @abhi @anusha-ragunathan

--- a/builder/builder-next/builder.go
+++ b/builder/builder-next/builder.go
@@ -37,6 +37,7 @@ func init() {
 type Opt struct {
 	SessionManager    *session.Manager
 	Root              string
+	NetnsRoot         string
 	Dist              images.DistributionServices
 	NetworkController libnetwork.NetworkController
 }

--- a/builder/builder-next/controller.go
+++ b/builder/builder-next/controller.go
@@ -90,7 +90,7 @@ func newController(rt http.RoundTripper, opt Opt) (*control.Controller, error) {
 		return nil, err
 	}
 
-	exec, err := newExecutor(root, opt.NetworkController)
+	exec, err := newExecutor(root, opt.NetnsRoot, opt.NetworkController)
 	if err != nil {
 		return nil, err
 	}

--- a/builder/builder-next/executor_unix.go
+++ b/builder/builder-next/executor_unix.go
@@ -3,41 +3,46 @@
 package buildkit
 
 import (
-	"fmt"
+	"os"
 	"path/filepath"
+	"strconv"
 	"sync"
 
 	"github.com/docker/libnetwork"
 	"github.com/moby/buildkit/executor"
 	"github.com/moby/buildkit/executor/runcexecutor"
 	"github.com/moby/buildkit/identity"
+	"github.com/moby/buildkit/solver/pb"
 	"github.com/moby/buildkit/util/network"
-	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
 )
 
 const networkName = "bridge"
 
-func newExecutor(root string, net libnetwork.NetworkController) (executor.Executor, error) {
-	// FIXME: fix bridge networking
-	_ = bridgeProvider{}
+func newExecutor(root, netnsRoot string, net libnetwork.NetworkController) (executor.Executor, error) {
+	networkProviders := map[pb.NetMode]network.Provider{
+		pb.NetMode_UNSET: &bridgeProvider{NetworkController: net, netnsRoot: netnsRoot},
+		pb.NetMode_HOST:  network.NewHostProvider(),
+		pb.NetMode_NONE:  network.NewNoneProvider(),
+	}
 	return runcexecutor.New(runcexecutor.Opt{
 		Root:              filepath.Join(root, "executor"),
 		CommandCandidates: []string{"docker-runc", "runc"},
-	}, nil)
+	}, networkProviders)
 }
 
 type bridgeProvider struct {
 	libnetwork.NetworkController
+	netnsRoot string
 }
 
-func (p *bridgeProvider) NewInterface() (network.Interface, error) {
+func (p *bridgeProvider) New() (network.Namespace, error) {
 	n, err := p.NetworkByName(networkName)
 	if err != nil {
 		return nil, err
 	}
 
-	iface := &lnInterface{ready: make(chan struct{})}
+	iface := &lnInterface{ready: make(chan struct{}), provider: p}
 	iface.Once.Do(func() {
 		go iface.init(p.NetworkController, n)
 	})
@@ -45,33 +50,13 @@ func (p *bridgeProvider) NewInterface() (network.Interface, error) {
 	return iface, nil
 }
 
-func (p *bridgeProvider) Release(iface network.Interface) error {
-	go func() {
-		if err := p.release(iface); err != nil {
-			logrus.Errorf("%s", err)
-		}
-	}()
-	return nil
-}
-
-func (p *bridgeProvider) release(iface network.Interface) error {
-	li, ok := iface.(*lnInterface)
-	if !ok {
-		return errors.Errorf("invalid interface %T", iface)
-	}
-	err := li.sbx.Delete()
-	if err1 := li.ep.Delete(true); err1 != nil && err == nil {
-		err = err1
-	}
-	return err
-}
-
 type lnInterface struct {
 	ep  libnetwork.Endpoint
 	sbx libnetwork.Sandbox
 	sync.Once
-	err   error
-	ready chan struct{}
+	err      error
+	ready    chan struct{}
+	provider *bridgeProvider
 }
 
 func (iface *lnInterface) init(c libnetwork.NetworkController, n libnetwork.Network) {
@@ -99,14 +84,26 @@ func (iface *lnInterface) init(c libnetwork.NetworkController, n libnetwork.Netw
 	iface.ep = ep
 }
 
-func (iface *lnInterface) Set(pid int) error {
+func (iface *lnInterface) Set(s *specs.Spec) {
 	<-iface.ready
 	if iface.err != nil {
-		return iface.err
+		return
 	}
-	return iface.sbx.SetKey(fmt.Sprintf("/proc/%d/ns/net", pid))
+	// attach netns to bridge within the container namespace, using reexec in a prestart hook
+	s.Hooks = &specs.Hooks{
+		Prestart: []specs.Hook{{
+			Path: filepath.Join("/proc", strconv.Itoa(os.Getpid()), "exe"),
+			Args: []string{"libnetwork-setkey", iface.sbx.ContainerID(), iface.provider.NetworkController.ID()},
+		}},
+	}
 }
 
-func (iface *lnInterface) Remove(pid int) error {
-	return nil
+func (iface *lnInterface) Close() error {
+	<-iface.ready
+	err := iface.sbx.Delete()
+	if iface.err != nil {
+		// iface.err takes precedence over cleanup errors
+		return iface.err
+	}
+	return err
 }

--- a/builder/builder-next/executor_windows.go
+++ b/builder/builder-next/executor_windows.go
@@ -10,7 +10,7 @@ import (
 	"github.com/moby/buildkit/executor"
 )
 
-func newExecutor(_ string, _ libnetwork.NetworkController) (executor.Executor, error) {
+func newExecutor(_, _ string, _ libnetwork.NetworkController) (executor.Executor, error) {
 	return &winExecutor{}, nil
 }
 

--- a/cmd/dockerd/daemon.go
+++ b/cmd/dockerd/daemon.go
@@ -288,6 +288,7 @@ func newRouterOptions(config *config.Config, daemon *daemon.Daemon) (routerOptio
 	bk, err := buildkit.New(buildkit.Opt{
 		SessionManager:    sm,
 		Root:              filepath.Join(config.Root, "buildkit"),
+		NetnsRoot:         filepath.Join(config.ExecRoot, "netns"),
 		Dist:              daemon.DistributionServices(),
 		NetworkController: daemon.NetworkController(),
 	})

--- a/vendor.conf
+++ b/vendor.conf
@@ -26,7 +26,7 @@ github.com/imdario/mergo v0.3.6
 golang.org/x/sync 1d60e4601c6fd243af51cc01ddf169918a5407ca
 
 # buildkit
-github.com/moby/buildkit 49906c62925ed429ec9174a0b6869982967f1a39
+github.com/moby/buildkit e1cd06ad6b74e4b747306c4408c451b3b6d87a89
 github.com/tonistiigi/fsutil b19464cd1b6a00773b4f2eb7acf9c30426f9df42
 github.com/grpc-ecosystem/grpc-opentracing 8e809c8a86450a29b90dcc9efbf062d0fe6d9746
 github.com/opentracing/opentracing-go 1361b9cd60be79c4c3a7fa9841b3c132e40066a7

